demos/embedded/src/main/java/org/eclipse/jetty/demos/JettyDemos.java

```diff
@@ -22,71 +24,75 @@
- * </ul>
+ * A utility test class to locate the Jetty Demo build contents.
+ * <p>
+ * Looking for content in the /demos/ directory.
+ * </p>
  */
-public class JettyHome
+public class JettyDemos
 {
-    private static final Logger LOG = LoggerFactory.getLogger(JettyHome.class);
-    public static final Path JETTY_HOME;
+    private static final Logger LOG = LoggerFactory.getLogger(JettyDemos.class);
+    private static final Path JETTY_DEMOS_DIR;
+    private static final String VERSION;
 
     static
     {
-        Path jettyHome = asDirectory(System.getProperty("jetty.home"));
-        LOG.debug("JettyHome(prop(jetty.home)) = {}", jettyHome);
-        if (jettyHome == null)
+        Path demosDir = asDirectory(System.getProperty("jetty.demos"));
+        LOG.debug("JettyDemos(prop(jetty.demos)) = {}", demosDir);
+        if (demosDir == null)
         {
-            jettyHome = asDirectory(System.getenv().get("JETTY_HOME"));
-            LOG.debug("JettyHome(env(JETTY_HOME)) = {}", jettyHome);
+            demosDir = asDirectory(System.getenv().get("JETTY_DEMOS"));
+            LOG.debug("JettyDemos(env(JETTY_DEMOS)) = {}", demosDir);
         }
 
-        Path distro = null;
-
-        if (jettyHome != null && Files.exists(jettyHome.resolve("start.jar")))
-        {
-            distro = jettyHome;
-        }
-
-        if (distro == null)
+        if (demosDir == null || !Files.exists(demosDir.resolve("pom.xml")))
         {
             try
             {
                 Path working = Paths.get(System.getProperty("user.dir"));
                 Path dir = null;
-                LOG.debug("JettyHome(prop(user.dir)) = {}", working);
+                LOG.debug("JettyDemos(prop(user.dir)) = {}", working);
                 while (dir == null && working != null)
                 {
-                    dir = asDirectory(working.resolve("jetty-home/target/jetty-home").toString());
-                    if (dir != null && Files.exists(dir.resolve("start.jar")))
+                    dir = asDirectory(working.resolve("demos").toString());
+                    if (dir != null && Files.exists(dir.resolve("pom.xml")))
                     {
-                        distro = dir;
+                        demosDir = dir;
                     }
                     // try one parent up
                     working = working.getParent();
                 }
                 if (LOG.isDebugEnabled())
-                    LOG.debug("JettyHome(working.resolve(...)) = {}", distro);
+                    LOG.debug("JettyDemos(working.resolve(...)) = {}", demosDir);
             }
             catch (Throwable th)
             {
-                LOG.warn("Unable to resolve Jetty Distribution location", th);
+                LOG.warn("Unable to resolve Jetty Demos location", th);
             }
         }
 
-        if (distro == null)
+        JETTY_DEMOS_DIR = demosDir;
+
+        String version = "unknown";
+        Path pomFile = demosDir.resolve("pom.xml");
```

Comment:
version can come from surefire/poms as a system property.
in pom just add

```
        <plugin>
          <groupId>org.apache.maven.plugins</groupId>
          <artifactId>maven-surefire-plugin</artifactId>
          ....
          <configuration>
            ....
            <systemPropertyVariables>
....
              <jetty.version>${project.version}</jetty.version>
......
            </systemPropertyVariables>
          </configuration>
```
then 
```
VERSION = System.getProperty("jetty.version")
```

Reply:
Keep in mind that this is the demos/embedded.

This surefire trickery doesn't work from the various IDEs that people use.

* Eclipse - doesn't work
* IntelliJ - sometimes works
* NetBeans - doesn't work
* Emacs / Vim - never works
* Eclipse Che - doesn't work
* BlueJ - doesn't work
* XCode - doesn't work

I don't like this solution for the VERSION entry in demos/embedded

Reply:
we do this for test-distributions for ages and never heard any issues...